### PR TITLE
Add Filter Feature with Transaction ID, Quote ID, Amount Type, and Transaction Type

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: packages/domain-lib/src/interfaces/infrastructure.ts [hunk 1]
```diff
@@ -33,41 +33,61 @@
...
+  getQuotesByBulkQuoteIdAndStatus(
+    id: string,
+    status: QuoteStatus
+  ): Promise<IQuote[]>;
+  searchQuotes(
+    transactionId?: string,
+    quoteId?: string,
+    amountType?: string,
+    transactionType?: string
+  ): Promise<IQuote[] | null>;
```
Comment (pedrosousabarreto) [inline packages/domain-lib/src/interfaces/infrastructure.ts:56]: When returning arrays we don't need to use null, if none where found we just return and empty array.
Remove the "| null" part of the response type

**Review by pedrosousabarreto — CHANGES_REQUESTED** [review]
Please address that comment I made.
Everything else looks good! Thank you.

File: packages/domain-lib/src/interfaces/infrastructure.ts (hunk 1 above)
Comment (ghost) [reply packages/domain-lib/src/interfaces/infrastructure.ts:56]: Thanks, Pedro! Here is the commit (7385a375d8b60f6933aab2070644492aa5325ded) for your requested changes.

Comment (pedrosousabarreto) [reply packages/domain-lib/src/interfaces/infrastructure.ts:56]: Looks good! thanks

**Comment by simeonoriko** [comment]
Congratulations @ryam-tw on your first Mojaloop commit!

**Comment by elnyry-sam-k** [comment]
congratulations @ryam-tw ! thank you!

**Pull request merged**